File: skillsphere-learning-service/src/main/java/com/infosis/nexus/learning/service/CertificateService.java
package com.infosis.nexus.learning.service;

import com.infosis.nexus.learning.dto.CertificateResponse;

import java.util.List;

public interface CertificateService {
    CertificateResponse generate(Long enrollmentId);
    CertificateResponse getById(Long id);
    List<CertificateResponse> getByEmployeeId(Long employeeId);
    List<CertificateResponse> getByCourseId(Long courseId);
    List<CertificateResponse> getAll();
}

File: frontend/src/services/certificateService.js
import api from './api';

const certificateService = {
    generate: (enrollmentId) => api.post(`/learning/certificates/generate/${enrollmentId}`),
    getAll: () => api.get('/learning/certificates'),
    getById: (id) => api.get(`/learning/certificates/${id}`),
    getByEmployeeId: (employeeId) => api.get(`/learning/certificates/employee/${employeeId}`),
    getByCourseId: (courseId) => api.get(`/learning/certificates/course/${courseId}`),
};

export default certificateService;